app/src/main/java/com/willowtree/vocable/settings/EditPresetsFragment.kt

```diff
@@ -0,0 +1,201 @@
+package com.willowtree.vocable.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.children
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import androidx.viewpager2.widget.ViewPager2
+import com.willowtree.vocable.BaseFragment
+import com.willowtree.vocable.R
+import com.willowtree.vocable.customviews.PointerListener
+import com.willowtree.vocable.databinding.FragmentEditPresetsBinding
+import com.willowtree.vocable.room.Phrase
+import java.lang.Math.ceil
+
+class EditPresetsFragment : BaseFragment() {
+
+    private var binding: FragmentEditPresetsBinding? = null
+    private var allViews = mutableListOf<View>()
+
+    private var maxPhrases = 1
+
+    private lateinit var editPhrasesViewModel: EditPhrasesViewModel
+    private lateinit var phrasesAdapter: EditPhrasesAdapter
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentEditPresetsBinding.inflate(inflater, container, false)
+
+        return binding?.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(resources) {
+            maxPhrases = getInteger(R.integer.max_edit_phrases)
```

Comment:
If you're only making one call with `resources` here, you can get rid of the `with` block:
`maxPhrases = resources.getInteger(R.integer.max_edit_phrases)`

Reply:
ah thank you I was trying to figure this one out haha